pmlc/util/strides.cc

```diff
@@ -0,0 +1,91 @@
+// Copyright 2020 Intel Corporation
+
+#include "pmlc/util/strides.h"
+
+#include <vector>
+
+#include "mlir/Analysis/AffineStructures.h"
+#include "pmlc/util/logging.h"
+
+namespace pmlc::util {
+
+StrideArray::StrideArray(unsigned numDims, int64_t offset)
+    : offset(offset), strides(numDims) {}
+
+StrideArray &StrideArray::operator*=(int64_t factor) {
+  offset *= factor;
+  for (auto &dim : strides)
+    dim *= factor;
+  return *this;
+}
+
+StrideArray &StrideArray::operator+=(const StrideArray &rhs) {
+  assert(strides.size() == rhs.strides.size() && "strides sizes much match");
+  offset += rhs.offset;
+  for (unsigned i = 0, e = strides.size(); i < e; ++i) {
```

Comment:
Maybe just a style thing, but why not just i < strides.size()?

Reply:
This is an LLVM-ism, their style guide prefers to compute the end once per loop instead of once per iteration.

https://llvm.org/docs/CodingStandards.html#don-t-evaluate-end-every-time-through-a-loop